charts/brigade/templates/api-deployment.yaml

```diff
@@ -23,6 +23,10 @@ spec:
       - name: {{ .Chart.Name }}
         image: "{{ .Values.api.registry }}/{{ .Values.api.name }}:{{ default .Chart.AppVersion .Values.api.tag }}"
         imagePullPolicy: {{ default "IfNotPresent" .Values.api.pullPolicy }}
+{{ if .Values.generic-gateway }}        
```

Comment:
Alas, `-` chars aren't allowed, so we'd need to go camelCase here...

While on the topic, I wonder if we do without the template conditional and just pass the value in, however it is set in the `values.yaml`:
```
        args:
          ["--generic-gateway", "{{ .Values.genericGateway }}"]
```
Also, I wonder if we should follow the precedent of appending `enabled` on the value for clarity, e.g. `genericGateway.enabled`, as is done for `rbac` ?  (Perhaps also the natural fit if we expect further properties to be added under the `genericGateway` config section.)

P.S., I tested the chart install via `helm install charts/brigade --dry-run --debug` to catch the aforementioned error, which also demonstrated a lack of test coverage in this repo, so I added a ticket to add, though in the future repo home of Brigade charts; see https://github.com/Azure/brigade-charts/issues/4.